### PR TITLE
Review Soliton C# styleguide

### PR DESCRIPTION
**Pull request by GowthamKarunakaran**
This PR is to get the style guide reviewed and address the review comments.

### CODE REVIEW

File: Readme.md [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+  - Private Methods
+  - Within each of the above elements group, Arrange in following order of access modifiers,
+    - public
+    - internal
+    - protected internal
+    - protected
+    - private
+
+---
+
```
Comment (Vasantha-Kumar-J) [inline Readme.md:170]: Can we talk about the Dispose Pattern(with template pattern) in this guideline?

File: Readme.md [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+- [D.1] Use XML documentations for class and its all elements inside.
+  - Take time and fill in with the good documentation as it would be the starting point for anyone who reads the code first.
+  - For complex logics, loops inside the methods, Add neccessary documentation explaining the need of the logic and how it works using double forward slash ('**_//_**')
+  - Documentation cannot be added for Namespaces.
+- [D.2] Do not have any commented code while pushing to repo.
+- [D.3] Refer the [link](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/xmldoc/) for XML documentation tags's usage and examples.
+
+```
+Tip:
+In VisualStudio, type forward slash('/') three times as '**_///_**'.
```
Comment (Vasantha-Kumar-J) [inline Readme.md:182]: Can we also add tips on referring to the other classes or methods using `<see cref="">`

File: Readme.md [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+---
+
+### Language Features
+
+- [L.1] Remove unused imports.
+- [L.2] Use primitive C# types instead of Framework Class Library (FCL) types.
+  - Primitive Types - ```sbyte, byte, short, ushort, int, uint, long, ulong, char, float, double, bool, decimal, object, string```.
+  - Framework Class Library (FCL) Types - ```SByte, Byte, Int16, UInt16, Int32, UInt32, Int64, UInt64, Char, Single, Double, Boolean, Decimal, Object, String```.
+- [L.3] Use either private fields (or) public properties. Do not use public fields.
+- [L.4] Use ```StringBuilder``` over ```string``` and operator ```+``` for multiple string concatenations as StringBuilder will have better performance.
```
Comment (Vasantha-Kumar-J) [inline Readme.md:194]: Can we add any distinction with respect to string interpolation here?

File: Readme.md [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+
+- [N.1] Identifiers must start with a **_letter_** (or) underscore **_' \_'_** in case of private fields.
+  > Identifier is the name you assign to a type (class, interface, struct, record, delegate, or enum), member, variable, or namespace.
+- [N.2] Casing:
+  - <table>
+        <thead>
+            <tr>
+                <th>Case</th>
+                <th>Type</th>
+                <th>Example</th>
```
Comment (Vasantha-Kumar-J) [inline Readme.md:26]: Can the content in the Example column be using the code syntax? ` ``` ``` `

File: Readme.md [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+            <tr>
+                <td><b>_c</b>amel<b>C</b>ase </br> (Underscore'<b>_</b>' + Camelcase)</td>
+                <td>Private Fields</td>
+                <td>public class Employee </br>{</br> private double_houseRentAllowance  = 10000; </br>..</br> }</br></td>
+            </tr>
+        </tbody>
+    </table>
+- [N.3] Use meaningful and descriptive names. Follow similar patterns for similar data types/ files.
+  - [N.3.1] **Boolean** – Start with helping verbs.
+    > ```is```Checked, ```can```Show, ```do```SelfCalibrate.
```
Comment (Vasantha-Kumar-J) [inline Readme.md:69]: ```do```SelfCalibrate looks like method name rather than a Boolean name

**Review by Vasantha-Kumar-J — COMMENTED** [review]
Looks good. Given a high level review and comments.

File: Readme.md (hunk 4 above)
Comment (GowthamKarunakaran) [reply Readme.md:26]: Could not add Codeblocks ` ``` ``` ` within a table in the markdown format.

File: Readme.md (hunk 5 above)
Comment (GowthamKarunakaran) [reply Readme.md]: Removed 'doSelfCalibrate' example.

File: Readme.md (hunk 4 above)
Comment (GowthamKarunakaran) [reply Readme.md:26]: Added as a backlog to format the examples inside table to same as ```csharp``` code in markdown format

File: Readme.md (hunk 3 above)
Comment (GowthamKarunakaran) [reply Readme.md]: Updated the guideline including string interpolation.
A detailed explanation will be added for string functions in a separate section. - Added as a Backlog

File: Readme.md (hunk 1 above)
Comment (GowthamKarunakaran) [reply Readme.md:168]: Added as a backlog to speak about Dispose Pattern

File: Readme.md [hunk 6]
```diff
@@ -0,0 +1,204 @@
...
+      }
+      ```
+
+- [C.2] Arrange the class elements in the following order,
+  - Constant fields
+  - Private fields
+  - Constructor
+  - Finalizers/Destructors
+  - Delegates/Events/Enums
+  - Public Properties
```
[the remark this answers is not in the document]
Comment (GowthamKarunakaran) [reply Readme.md]: Updated the modifiers order including combinations 'private protected'

File: Readme.md [hunk 7]
```diff
@@ -0,0 +1,204 @@
...
+This contains the coding conventions/style guides for all C# code we develop. This document is constantly evolving. New rules might be added over time.
+The guidelines have been categorized into following sections:
+
+- [Soliton C# Style Guide](#soliton-c-style-guide)
+  - [Coding Conventions](#coding-conventions)
+    - [Naming](#naming)
+    - [Code Organization](#code-organization)
+    - [Documentation](#documentation)
+    - [Language Features](#language-features)
+---
```
[the remark this answers is not in the document]
Comment (GowthamKarunakaran) [reply Readme.md:12]: Added as a backlog item

**Pull request closed without merging**